Packs/EWS/Integrations/EWSO365/EWSO365.py

```diff
@@ -0,0 +1,1765 @@
+    else:
+        raise Exception("Failed mark-item-as-junk with error: " + ews_result)
+
+    readable_output = tableToMarkdown("Mark item as junk", mark_as_junk_result)
+    output = {CONTEXT_UPDATE_EWS_ITEM: mark_as_junk_result}
+    return readable_output, output, mark_as_junk_result
+
+
+def get_items_from_folder(
+    client: EWSClient,
+    folder_path,
+    limit=100,
+    target_mailbox=None,
+    is_public=None,
+    get_internal_item="no",
+):
+    account = client.get_account(target_mailbox)
+    limit = int(limit)
+    get_internal_item = get_internal_item == "yes"
+    is_public = client.is_default_folder(folder_path, is_public)
+    folder = client.get_folder_by_path(folder_path, account, is_public)
+    qs = folder.filter().order_by("-datetime_created")[:limit]
+    items = get_limited_number_of_messages_from_qs(qs, limit)
+    items_result = []
+
+    for item in items:
+        item_attachment = parse_item_as_dict(
+            item, account.primary_smtp_address, camel_case=True, compact_fields=True
+        )
+        for attachment in item.attachments:
+            if (
+                get_internal_item
+                and isinstance(attachment, ItemAttachment)
+                and isinstance(attachment.item, Message)
+            ):
+                # if found item attachment - switch item to the attchment
+                item_attachment = parse_item_as_dict(
+                    attachment.item,
+                    account.primary_smtp_address,
+                    camel_case=True,
+                    compact_fields=True,
+                )
+                break
+        items_result.append(item_attachment)
+
+    hm_headers = [
+        "sender",
+        "subject",
+        "hasAttachments",
+        "datetimeReceived",
+        "receivedBy",
+        "author",
+        "toRecipients",
+        "id",
+    ]
+    readable_output = tableToMarkdown(
+        "Items in folder " + folder_path, items_result, headers=hm_headers
+    )
+    output = {CONTEXT_UPDATE_EWS_ITEM: items_result}
+    return readable_output, output, items_result
+
+
+def get_items(client: EWSClient, item_ids, target_mailbox=None):
+    item_ids = argToList(item_ids)
+    account = client.get_account(target_mailbox)
+    items = client.get_items_from_mailbox(account, item_ids)
+    items = [x for x in items if isinstance(x, Message)]
+    items_as_incidents = [parse_incident_from_item(x) for x in items]
+    items_to_context = [
+        parse_item_as_dict(x, account.primary_smtp_address, True, True) for x in items
+    ]
+    readable_output = tableToMarkdown(
+        "Get items", items_to_context, ITEMS_RESULTS_HEADERS
+    )
+    output = {
+        CONTEXT_UPDATE_EWS_ITEM: items_to_context,
+        "Email": [email_ec(item) for item in items],
+    }
+    return readable_output, output, items_as_incidents
+
+
+def get_folder(client: EWSClient, folder_path, target_mailbox=None, is_public=None):
```

Comment:
target_mailbox is not used

Reply:
nice catch, this was missed in the migration - it's an optional value. Added relevant code.